### DISCUSSION TITLE
compat.PyTorch2RiverClassifier problem

### DISCUSSION THREAD

**Discussion by ghost** [online-ml/river#990] [Q&A] [answered]
Hi!

I am struggling with **compat.PyTorch2RiverClassifier**, (A river classifier that integrates neural Networks from PyTorch) could anyone help me please?
I have a splipt with my resnet18 model and its pre-trained weights in pytorch and i would like to use river, but it seems like is impossible to make an inference! I define the class model as always in pytorch, loaded weights but i dont know how to use:

**class definition**
class Resnet18_Model(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.model=models.resnet18(pretrained=True)
        for param in self.model.parameters():
            param.requires_grad = False
        num_ftrs = self.model.fc.in_features
        self.model.fc=nn.Sequential
            nn.Linear(num_ftrs, 2),  #2 possible output 
            nn.Softmax(dim=1)
        ) 
    def forward(self, xb):
        return self.model(xb)

def build_classifier(n_features):
    return Resnet18_Model()

**pytorch to river conversion**
rivermodel = river.compat.PyTorch2RiverClassifier(build_fn = build_classifier,
                                    loss_fn = nn.CrossEntropyLoss)

**image inference** 
rivermodel.predict_proba_one(image)

Image is my tensor (3d image) but i dont know how to give it to the predict_proba_one function. I can read in docs that it must be a dict() but how? Should i flatten the 3d tensor? 

Where shoud i instance the model?  (model=Resnet18_model())?

Any help would be appreciated.

**Comment by MaxHalford** [comment, marked as answer]
Could you check out [river-torch](https://github.com/online-ml/river-torch)? We're going to deprecate the PyTorch stuff in the compat module.

**Reply by ghost** [reply]
Hi @MaxHalford, thank you for your help, but i have the same problem here,....how to send my 3d tensor image to the predict_one(x) function? x must be a dict, right?

Thank you

**Reply by MaxHalford** [reply]
Yes we only work with dictionaries, not tensors.

**Reply by ghost** [reply]
OK, then I assume that this library is not designed to work with images (tensors), as we can do in pytorch. As I see this is more oriented to ML aplications instead of DL aplications and thats why your example datasets are in .csv format.  right? If don't, I would appreciate if you can share an example in which I could see how to make inference with a tensor.

Thanks for your support.

**Reply by MaxHalford** [reply]
Indeed we cater to classic ML problems :)
